src/service/ampdoc-impl.js

```diff
@@ -223,6 +285,30 @@ export class AmpDocSingle extends AmpDoc {
   getUrl() {
     return this.win.location.href;
   }
+
+  /** @override */
+  getBody() {
+    return this.win.document.body;
+  }
+
+  /** @override */
+  onBody(callback) {
+    if (this.win.document.body) {
+      callback(this.win.document.body);
```

Comment:
can we do this in a microtask to avoid the method behaving synchronously sometimes and asynchronously other times.